pandas/core/internals/managers.py

```diff
@@ -441,11 +441,27 @@ def fillna(self: T, value, limit, inplace: bool, downcast) -> T:
     def astype(self: T, dtype, copy: bool = False, errors: str = "raise") -> T:
         return self.apply("astype", dtype=dtype, copy=copy, errors=errors)
 
-    def convert(self: T, copy: bool) -> T:
-        return self.apply(
+    def convert(self: T, copy: bool | None) -> T:
+        if not copy and using_copy_on_write():
+            copy = False
+        elif copy is None:
+            copy = True
+
+        if self.is_single_block:
+            original_blocks = [self.blocks[0]] * self.shape[0]
```

Comment:
Why is the `* self.shape[0]` needed?

Reply:
We are splitting blocks so we need the original blocks references * the number of columns

Reply:
Only in case of object dtype, right? (although I suppose this list multiplication is not necessarily expensive, so maybe not worth to avoid doing when not needed)

Reply:
No it's not that easy unfortunately,

Non object blocks are never copied, hence we need the references especially in this case. While object blocks might get copied (depends on the inference), so we might need the references there.